### PR TITLE
AnimInverseKinematics and AnimController node

### PR DESCRIPTION
**Pull request by hyperlogic**
* When AnimGraph is enabled the root IK node is ALWAYS enabled.  Even when there are no hand controllers.
* The IK solution is full body, no constraints are in play.  This results in very flexible spine and elbow joints.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3149baeefc65a03fc048a86c746e3a72c95e6776 available at: https://gist.github.com/77607a47c6403d3b7479<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5768.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5768.dmg

File: tests/animation/src/AnimInverseKinematicsTests.cpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
+//
+//  AnimInverseKinematicsTests.cpp
```
Comment (howard-stearns) [inline tests/animation/src/AnimInverseKinematicsTests.cpp]: These don't compile yet.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a9838fd0d97cdf349399f9c7d9849aa595d5729 available at: https://gist.github.com/77607a47c6403d3b7479<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5768.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5768.dmg

File: libraries/animation/src/AnimationCache.h [hunk 2]
```diff
@@ -12,9 +12,9 @@
 #ifndef hifi_AnimationCache_h
 #define hifi_AnimationCache_h
 
-#include <QRunnable>
-#include <QScriptEngine>
-#include <QScriptValue>
+#include <QtCore/QRunnable>
```
Comment (howard-stearns) [inline libraries/animation/src/AnimationCache.h]: Why these changes (and for ResourceCache.h)? Does this PR actually need changes to resource headers? Merging issue? gcc compiler fixes?

File: libraries/networking/src/ResourceCache.h [hunk 3]
```diff
@@ -12,17 +12,17 @@
...
-#include <QNetworkReply>
-#include <QNetworkRequest>
-#include <QObject>
-#include <QPointer>
-#include <QSharedPointer>
-#include <QUrl>
-#include <QWeakPointer>
-#include <QReadWriteLock>
-#include <QQueue>
+#include <QtCore/QHash>
```
Comment (howard-stearns) [inline libraries/networking/src/ResourceCache.h]: See comments for AnimationCache.h.

**Comment by howard-stearns** [comment]
Great integration! It doesn't look like anything broke. I've checked the cartesian product of:
- with and without rig animation
- with and without avatar update thread
- mac and windows
- hydra
Plus HMD on mac only, in both standing and sitting mode. (The standing HMD was a bit wonky, but I think it was before.)
Plus no crashes and expected behavior for anim-graph in main thread. (Might be a separate thread anim-graph conflict that I'll have to poke at, but I wouldn't let that stop a merge.)

File: libraries/animation/src/AnimationCache.h (hunk 2 above)
Comment (hyperlogic) [inline libraries/animation/src/AnimationCache.h]: These changes were made by @AndrewMeadows, not sure perhaps they were made for Linux?

File: tests/animation/src/AnimInverseKinematicsTests.cpp (hunk 1 above)
Comment (hyperlogic) [inline tests/animation/src/AnimInverseKinematicsTests.cpp]: Good catch, I haven't updated the tests.
